marklogic-rdf4j/src/test/java/com/marklogic/semantics/rdf4j/MarkLogicRepositoryConnectionTest.java

```diff
@@ -662,6 +660,9 @@ public void testHasStatementBNode() throws Exception
         Statement st1 = vf.createStatement(alice, name, alicesName);
         conn.add(st1, context1);
         Assert.assertTrue(conn.hasStatement(null, null, alicesName, false));
+
+        BNode markName = vf.createBNode();
+        Assert.assertFalse(conn.hasStatement(null, null, markName, false));
```

Comment:
The only way to test this function, which is almost an absurd use case, is to test for this:

```
BNode b = vh.createBNode();
Assert.assertTrue(conn.hasStatement(b, null, b, false));

in some triple store where there's a pattern like    :a :b :a, where subject and object are the same.